pyprep/utils.py

```diff
@@ -20,6 +21,28 @@ def _intersect(list1, list2):
     return list(set(list1).intersection(set(list2)))
 
 
+def mat_quantile(arr, q, axis=None):
+    # MATLAB calculates quantiles using different logic than Numpy: whereas
+    # Numpy treats the provided values as a whole population, MATLAB treats
+    # the input array as a sample from a population of unknown size.
+    # This function adjusts the quantiles to mimic MATLAB's behaviour.
+    q = np.asarray(q, dtype=np.float64)
+    n = len(arr)
+    q_adj = ((q - 0.5) * n / (n - 1)) + 0.5
+    return np.quantile(arr, np.clip(q_adj, 0, 1), axis=axis)
+
+
+def mat_iqr(arr, axis=None):
+    # MATLAB calculates IQRs using different logic than Numpy: whereas
+    # Numpy treats the provided values as a whole population, MATLAB treats
+    # the input array as a sample from a population of unknown size.
+    # This function adjusts the IQR to mimic MATLAB's behaviour.
```

Comment:
The comments you make here could well go into a `Notes` section of a numpydoc string

```python
"""
...

Notes
-----
MATLAB calculates quantiles using different logic than Numpy: where ....
"""
```